services/catalog/rs_server_catalog/user_catalog.py

```diff
@@ -1040,37 +1014,55 @@ async def manage_delete_response(self, response: StreamingResponse, user: str) -
         if "deleted collection" in response_content:
             response_content["deleted collection"] = response_content["deleted collection"].removeprefix(f"{user}_")
         # delete the s3 files as well
-        self.delete_s3_files()
+        delete_s3_files(self.s3_files_to_be_deleted)
+        self.s3_files_to_be_deleted.clear()
         return JSONResponse(response_content)
 
-    async def build_filelist_to_be_deleted(self, request):
+    async def build_fileslist_to_be_deleted(self, request):
```

Comment:
In fact the correct form is "file list" not "files list" :) https://english.stackexchange.com/questions/164714/item-queue-vs-items-queue-files-list-vs-file-list

Reply:
Indeed, i set it right the first time :) I will correct it